src/shared/socket-netlink.c

```diff
@@ -235,6 +235,70 @@ int socket_address_parse_and_warn(SocketAddress *a, const char *s) {
         return 0;
 }
 
+int socket_address_parse_ip(SocketAddress *a, const char *s) {
```

Comment:
wouldn't it be simpler to make this a wrapper around socket_address_parse() and just after it completed validate the right family is set?

Reply:
I dropped is completely and tweaked the parser so to reuse it .